packages/web-app-files/src/components/AppBar/AppBar.vue

```diff
@@ -20,83 +20,109 @@
+              :aria-label="newButtonAriaLabel"
+              variation="primary"
+              appearance="filled"
+              :disabled="isNewBtnDisabled"
+            >
+              <oc-icon name="add" />
+              <translate>New</translate>
+            </oc-button>
+            <oc-drop
+              drop-id="new-file-menu-drop"
+              toggle="#new-file-menu-btn"
+              mode="click"
+              close-on-click
+              :options="{ delayHide: 0 }"
+            >
+              <ul class="uk-list">
+                <li>
+                  <file-upload
+                    :path="currentPath"
+                    :headers="headers"
+                    @success="onFileSuccess"
+                    @error="onFileError"
+                    @progress="onFileProgress"
+                  />
+                </li>
+                <li v-if="checkIfBrowserSupportsFolderUpload">
+                  <folder-upload
+                    v-if="!isIE11()"
+                    :root-path="currentPath"
+                    :path="currentPath"
+                    :headers="headers"
+                    @success="onFileSuccess"
+                    @error="onFileError"
+                    @progress="onFileProgress"
+                  />
+                </li>
+                <li>
+                  <div>
+                    <oc-button
+                      id="new-folder-btn"
+                      appearance="raw"
+                      class="uk-width-1-1"
+                      justify-content="left"
+                      @click="showCreateResourceModal"
+                    >
+                      <oc-icon name="create_new_folder" />
+                      <translate>New folder…</translate>
+                    </oc-button>
+                  </div>
+                </li>
+                <li v-for="(newFileHandler, key) in newFileHandlers" :key="key">
+                  <div>
+                    <oc-button
+                      appearance="raw"
+                      justify-content="left"
+                      :class="['new-file-btn-' + newFileHandler.ext, 'uk-width-1-1']"
+                      @click="
+                        showCreateResourceModal(false, newFileHandler.ext, newFileHandler.action)
+                      "
+                    >
+                      <oc-icon :name="newFileHandler.icon || 'save'" />
+                      <span>{{ newFileHandler.menuTitle($gettext) }}</span>
+                    </oc-button>
+                  </div>
+                </li>
+              </ul>
+            </oc-drop>
+          </template>
+          <size-info v-if="selectedFiles.length > 0" class="oc-mr-s uk-visible@l" />
+          <batch-actions />
+        </div>
+        <oc-button
+          id="files-view-options-btn"
+          key="files-view-options-btn"
+          v-oc-tooltip="newButtonTooltip"
+          data-test-id="files-view-options-btn"
+          :aria-label="newButtonAriaLabel"
```

Comment:
Needs its own computed property + translation to give context for a11y reasons :)